### PR TITLE
Fix label_id_offset inconsistencies (obj detection)

### PR DESCRIPTION
**Pull request by korrawat**
The ["using your own dataset" tutorial](https://github.com/tensorflow/models/blob/master/object_detection/g3doc/using_your_own_dataset.md) says "Label maps should always start from id 1." However, the current code still has some inconsistencies.

Right now, the default value for `label_id_offset` in `eval_util.py`'s [`evaluate_detection_results_pascal_voc` method](https://github.com/tensorflow/models/blob/master/object_detection/eval_util.py#L111-L125) is 0 (instead of 1). When I run an evaluation using a pre-trained model on the PASCAL VOC, for example, `num_classes` will be 21 instead of 20, giving me the following warning.

```bash
WARNING:root:The following classes have no ground truth examples: 0
```
Changing the default value of `label_id_offset` to 1 will fix it.

### Questions
Should the comment in L111 ("Pascal VOC evaluator assumes foreground index starts from zero.") also change from "zero" to "one"?

By the way, what is allowed as the `categories`?
- At least, the ids should start from 1. The `label_id_offset` can allow for other cases.
- Maybe they should also be in an increasing order and consecutive? I'm not sure whether anything will break if this condition is not satisfied (e.g. the label map is 1=cat1, 5=cat5, 3=cat3). It seems like some codes are making these assumptions?
If these should be satisfied, maybe some checks can be added in `label_map_util.py`'s [`_validate_label_map` method](https://github.com/tensorflow/models/blob/master/object_detection/utils/label_map_util.py#L25).

Possibly related issues: #1856, #1936 

@derekjchow Could you please take a look? Thank you!

### CODE REVIEW

**Comment by tombstone** [comment]
@korrawat . Thanks for the PR. Taking a look.

**Comment by pkdogcom** [comment]
@tombstone Can you merge this commit to the master branch? I believe many users are experiencing this issue when training with with their own datasets. Thanks.

**Comment by tombstone** [comment]
@pkdogcom  @korrawat,  we ended up refactoring some evaluation module internally to remove inconsistencies. I will be merging it shortly. Apologies for the delay.

**Comment by jingyibo123** [comment]
Hello everyone,
Could anyone take a look at this [question](https://stackoverflow.com/questions/47099606/eval-py-fails-to-identify-classes-in-test-tfrecord), could it be caused by this issue?
